### ISSUE TITLE
`bundle exec rake new_release` fails

### ISSUE THREAD

**Issue by mokagio** [wordpress-mobile/release-toolkit#385]
I just opened the PR for 5.0.1, #384, and run into an issue on my M1 Max. When I run the `bundle exec rake new_release` I got the following error:

```
➜ br new_release
>>> Current version is: 5.0.0
>>> Pending CHANGELOG:
| ### Bug Fixes
|
| - Fix metadata length computation logic [[#383](https://github.com/wordpress-mobile/release-toolkit/pull/383)]
|

New version to release? [default: 5.0.1]

git checkout -b release/5.0.1
Switched to a new branch 'release/5.0.1'
>>> Update `VERSION` constant in `version.rb`...
bundle install --quiet
Bundler could not find compatible versions for gem "nokogiri":
  In Gemfile:
    fastlane-plugin-wpmreleasetoolkit was resolved to 5.0.1, which depends on
      nokogiri (~> 1.11)

Could not find gem 'nokogiri (~> 1.11)' with platform 'ruby', which is required by gem 'fastlane-plugin-wpmreleasetoolkit', in locally installed gems.

The source contains the following gems matching 'nokogiri (~> 1.11)':
  * nokogiri-1.13.6-arm64-darwin
  * nokogiri-1.13.7-arm64-darwin
rake aborted!
Command failed with status (6): [bundle install --quiet...]
/Users/gio/Developer/a8c/release-toolkit/Rakefile:89:in `update_version_constant'
/Users/gio/Developer/a8c/release-toolkit/Rakefile:57:in `block in <top (required)>'
/Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/Users/gio/.rbenv/versions/2.7.4/bin/bundle:23:in `load'
/Users/gio/.rbenv/versions/2.7.4/bin/bundle:23:in `<main>'
Tasks: TOP => new_release
(See full trace by running task with --trace)
```

`bundle install` works fine on the same machine.

I had to switch to my Intel Mac for the automation to succeed.

**Comment by mokagio** [comment]
Now that I think about it, I wonder if running `rake` via `bundle exec` had something to do with the issue.

**Comment by mokagio** [comment]
> Now that I think about it, I wonder if running `rake` via `bundle exec` had something to do with the issue.

Our `Rakefile` requires using `bundle exec`: https://github.com/wordpress-mobile/release-toolkit/blob/24734703550b080baf197acea1894df8146da81a/Rakefile#L4-L9

**Comment by mokagio** [comment]
I'm getting the same error on my Intel machine now...

```
➜ br new_release
>>> Current version is: 5.0.0
>>> Pending CHANGELOG:
| ### New Features
|
| * Allow using the `BUILDKITE_API_TOKEN` environment variable for the `buildkite_trigger_build` action. [#386]
|
| ### Bug Fixes
|
| - Fix metadata length computation logic [[#383](https://github.com/wordpress-mobile/release-toolkit/pull/383)]
|

New version to release? [default: 5.1.0]

git checkout release/5.1.0
Switched to branch 'release/5.1.0'
>>> Update `VERSION` constant in `version.rb`...
bundle install --quiet
Bundler could not find compatible versions for gem "nokogiri":
  In Gemfile:
    fastlane-plugin-wpmreleasetoolkit was resolved to 5.1.0, which depends on
      nokogiri (~> 1.11)

Could not find gem 'nokogiri (~> 1.11)' with platform 'ruby', which is required by gem 'fastlane-plugin-wpmreleasetoolkit', in locally installed gems.

The source contains the following gems matching 'nokogiri (~> 1.11)':
  * nokogiri-1.13.7-x86_64-darwin
rake aborted!
Command failed with status (6): [bundle install --quiet...]
/Users/gio/Developer/a8c/release-toolkit/Rakefile:89:in `update_version_constant'
/Users/gio/Developer/a8c/release-toolkit/Rakefile:57:in `block in <top (required)>'
/Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/Users/gio/.rbenv/versions/2.7.4/bin/bundle:23:in `load'
/Users/gio/.rbenv/versions/2.7.4/bin/bundle:23:in `<main>'
Tasks: TOP => new_release
(See full trace by running task with --trace)
```

**Comment by AliSoftware** [comment]
Ended up having a similar issue during the release of `5.2.0` (only [on CI here](https://buildkite.com/automattic/release-toolkit/builds/679#01826f7f-9ac4-4359-864a-98f25a35f0e5) though, not locally), so decided to land your workaround #387 after all. 

This should hopefully fix this once and for all 🤞  So I'll close this issue now. Feel free to re-open if you encounter the issue again during a subsequent release though.
